src/llmcompressor/transformers/compression/sparsity_config.py

```diff
@@ -67,13 +67,14 @@ def infer_sparsity_structure(model: Optional[Module] = None) -> str:
         if model and sparsity_structure is None:
             sparsity_structure = infer_sparsity_structure_from_model(model)
 
-        return sparsity_structure or "unstructured"
+        return SparsityStructure(sparsity_structure).value
 
     @staticmethod
     def from_pretrained(
         model: Module,
         state_dict: Optional[Dict[str, Tensor]] = None,
         compress: bool = False,
+        is_marlin: bool = False,
```

Comment:
To make this more generic, why not pass in the quantization config? We will for sure have different compression formats which affect sparsity in the future

Reply:
Yeah we should make this generic as marlin will likely not be the only case